src/backend/distributed/planner/distributed_planner.c

```diff
@@ -192,6 +193,14 @@ distributed_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
 			AdjustPartitioningForDistributedPlanning(rangeTableList,
 													 setPartitionedTablesInherited);
 		}
+		else
```

Comment:
I wasn't expecting to see that our logic is in `needsDistributedPlanning  == false`. Instead, I thought we'd do it in `RouterPlanner`, or even `FastPathRouterPlanner`. 

I'm OK with merging like this, but we should probably discuss a follow-up PR to move this to somewhere else.